src/kernel/mconfig_dot.ml

```diff
@@ -129,6 +129,34 @@ end = struct
     | Dot_merlin -> "dot-merlin-reader"
     | Dune -> "dune"
 
+  let find_exe = function
+    | Dot_merlin ->
+      (* 1. If DOT_MERLIN_READER_EXE is defined, then use its value for the
+            dot-merlin-reader exe
+         2. If not, look in the same directory as the merlin executable for a
+            dot-merlin-reader.
+         3. If not, fallback to using whatever one is on the PATH. *)
+      let get_from_env_var = lazy (Sys.getenv_opt "DOT_MERLIN_READER_EXE") in
+      let get_from_same_dir_as_merlin_exe =
+        lazy
+          (let merlin_exe = Sys.executable_name in
```

Comment:
I'm not totally sure but I think you may need to do a Unix.realpath on this, to make sure it's absolute and that we're not running via a symlink.  Can't hurt, anyway

Reply:
done